src/stepfunctions/steps/states.py

```diff
@@ -701,8 +701,9 @@ def visit(self, state):
 
     def _validate_next_step_params(self, params, step_output):
         for k, v in params.items():
-            if isinstance(v, StepInput) and v.name and not step_output.contains(v):
-                return False, k
+            if isinstance(v, StepInput):
+                if v.name and not step_output.contains(v):
```

Comment:
The proper fix should not be checking the `v.name` but to verify if `v` is the `step_output`.

For instance, if this is the input.
```json
{
    "ParamI": StepTwo.output()
}
```

This will make `not step_output.contains(v)` to be False because `StepTwo.output().contains(StepTwo.output()) == False`. Because we don't want to fail the validation, we only need to check if the output itself is being passed.

Reply:
Not quite sure what you mean here, can you explain?